include/aikido/perception/ObjectDatabase.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef AIKIDO_PERCEPTION_OBJECT_DATABASE_HPP_
+#define AIKIDO_PERCEPTION_OBJECT_DATABASE_HPP_
+
+#include <stdexcept>
+#include <dart/common/LocalResourceRetriever.hpp>
+#include <dart/dart.hpp>
+#include <aikido/io/CatkinResourceRetriever.hpp>
+#include <aikido/io/yaml.hpp>
+
+namespace aikido {
+namespace perception {
+
+/// Instantiation of ObjectDatabase that reads of JSON file containing the
+/// information that maps object keys to the object names and resources.
+
+/// The JSON file should have a map with object keys.
+/// Each such key points to a nested map, where the keys are
+/// 'resource' and 'name'.
+/// The values for each of the nested keys for a particular
+/// object key are to be returned to the calling method
+/// via the callback.
+///
+/// Here is an example entry in a JSON file:
+/// \code
+/// "obj_key": {
+///    "resource": "package://pr_ordata/data/objects/obj_filename.urdf",
+///    "name": "obj_name"
+///  }
+/// \endcode
+
+class ObjectDatabase
+{
+public:
+  /// Construct a \c ObjectDatabase that uses \c ResourceRetriever to
+  /// load configuration data from a JSON file at URI \c configDataURI.
+  /// \param[in] resourceRetriever The pointer to obtain the configuration file
+  /// \param[in] configDataURI The URI for the configuration information file
+  ObjectDatabase(
+      const dart::common::ResourceRetrieverPtr& resourceRetriever,
+      dart::common::Uri configDataURI);
+
+  virtual ~ObjectDatabase() = default;
+
+  bool getObjectByKey(
```

Comment:
Can you document this? In particular, it seems from the code that it should have `\param[out] obj_name` and `\param[out] obj_resource`.